src/LibraryManager.Vsix/UI/Models/InstallDialogViewModel.cs

```diff
@@ -338,8 +340,56 @@ private bool CanInstallPackage()
                 return false;
             }
 
+            return true;
+        }
+
+        public string ErrorMessage
+        {
+            get { return _errorMessage; }
+            set { Set(ref _errorMessage, value); }
+        }
+
+        public bool IsLibraryInstallationStateValid()
+        {
+            LibraryInstallationState libraryInstallationState = new LibraryInstallationState
+            {
+                LibraryId = PackageId,
+                ProviderId = SelectedProvider.Id,
+                DestinationPath = InstallationFolder.DestinationFolder,
+                Files = SelectedFiles?.ToList()
+            };
+
+            if (libraryInstallationState == null)
+            {
+                ErrorMessage = PredefinedErrors.UnknownError().Message;
+                return false;
+            }
+
+            IList<IError> errors = new List<IError>();
+
+            Shell.ThreadHelper.JoinableTaskFactory.Run(async () =>
+            {
+                ILibraryOperationResult libraryOperationResult = await libraryInstallationState.IsValidAsync(SelectedProvider).ConfigureAwait(false);
+                errors = libraryOperationResult.Errors;
+            });
+
+            ErrorMessage = string.Empty;
+
+            if (errors.Count > 0)
+            {
+                ErrorMessage = errors.First().Message;
```

Comment:
>First() [](start = 38, length = 7)

[0] is better on a list

Reply:
Updated in iteration 7

---
In reply to: [201159855](https://github.com/aspnet/LibraryManager/pull/293#discussion_r201159855) [](ancestors = 201159855)